quesma/quesma/ui/login.go

```diff
@@ -0,0 +1,60 @@
+// Copyright Quesma, licensed under the Elastic License 2.0.
+// SPDX-License-Identifier: Elastic-2.0
+
+package ui
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"quesma/elasticsearch"
+)
+
+func (qmc *QuesmaManagementConsole) generateLoginForm() []byte {
+	buffer := newBufferWithHead()
+	buffer.Html(`<html>`)
+	buffer.Html(`<body>`)
+	buffer.Html(`<div class="login-screen">`)
+	buffer.Html(`<div class="login-form">`)
+	buffer.Html(`<h2>Login</h2>`)
+	buffer.Html(`<p style="color: #ccc;">Log in to Quesma admin console using your Elasticsearch credentials</p>`)
+	buffer.Html(`<form action="/login-with-elasticsearch" method="post">`)
+	buffer.Html(`<label for="username">Username:</label>`)
+	buffer.Html(`<input type="text" id="username" name="username" placeholder="Enter your Elasticsearch username">`)
+	buffer.Html(`<label for="password">Password:</label>`)
+	buffer.Html(`<input type="password" id="password" name="password" placeholder="Enter your Elasticsearch password">`)
+	buffer.Html(`<input type="submit" value="Login">`)
+	buffer.Html(`</form>`)
+	buffer.Html(`</div>`)
+	buffer.Html(`</div>`)
+	buffer.Html(`</body>`)
+	buffer.Html(`</html>`)
+	return buffer.Bytes()
+}
+
+func (qmc *QuesmaManagementConsole) HandleElasticsearchLogin(writer http.ResponseWriter, req *http.Request) {
+	if req.Method == http.MethodGet {
```

Comment:
We should check if the user is logged in already here.

Reply:
https://github.com/QuesmaOrg/quesma/pull/836/commits/9bff908d6cddb3483472e0e4cdbdf3cd44f81c51